src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -351,20 +463,46 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
             // instead be the second mask. If so, skip the entire first mask and drain ASCII bytes
             // from the second mask.
 
-            if (currentMask == 0)
+            if (Sse2.IsSupported)
             {
-                pBuffer += SizeOfVector128;
-                currentMask = secondMask;
+                if (!ContainsNonAsciiByte_Sse2(currentSseMask))
+                {
+                    pBuffer += SizeOfVector128;
+                    currentSseMask = secondSseMask;
+                }
+            }
+            else if (AdvSimd.IsSupported)
+            {
+                if (!ContainsNonAsciiByte_AdvSimd(currentAdvSimdIndex))
+                {
+                    pBuffer += SizeOfVector128;
+                    currentAdvSimdIndex = secondAdvSimdIndex;
+                }
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
             }
-
         FoundNonAsciiDataInCurrentMask:
 
             // The mask contains - from the LSB - a 0 for each ASCII byte we saw, and a 1 for each non-ASCII byte.
```

Comment:
This comment applies to SSE2 case. Can you move it inside the `if (Sse2.IsSupported)` and have a separate comment for `AdvSimd.IsSupported` case?